examples/nft/EntryToken.xml

```diff
@@ -105,42 +105,34 @@
     </ts:attribute-type>
     <ts:attribute-type id="expired" syntax="1.3.6.1.4.1.1466.115.121.1.7"> <!-- boolean -->
       <ts:origins>
-        <ts:ethereum function="isExpired" contract="EntryToken">
+        <ts:ethereum function="isExpired" contract="EntryToken" as="bool">
           <ts:inputs>
```

Comment:
Should this not be:

<ts:ethereum function="isExpired" contract="EntryToken" as="uint">

so that the uint type is used for the mapping look up which is 0 or 1. At some point you'll want to use a mapping to return more than 2 choices so you can re-use the same syntax for consistency.

Reply:
See issue #127

Reply:
We still support `as="uint"` especially if you want to use a mapping like you said. For example if we had wanted to do this:

```
<ts:attribute-type as="utf8" id="locationName" syntax="1.3.6.1.4.1.1466.115.121.1.15">
    <ts:ethereum function="getLocationCode" contract="EntryToken" as="uint">
```

where the smart contract function `getLocationCode()` returns a uint from a hardcoded list and we want to convert it to a string for the attribute.

But in this case, it is not necessary to have a mapping because the syntax says it's a `Boolean` implying exactly 2 states, `true` and `false` in the native (Java/Swift, etc) language. (notice that the mapping tag was removed).

Reply:
then we remove the mapping because you explicitly declare the output as Boolean? I'm ok with that system too. 

But auto-translating a return type of Boolean back into int to select between a true or false mapping seems like a bad choice.

Reply:
Ok I didn't see the mapping block was also removed! I should view the complete changed file instead of the diff.

Reply:
I don't have an opinion on whether the mapping rule for bool type should be left out, but I do note that for `<mapping>`, the `as="xxx"` bit is not always uint. think someone who wants to retrofit an old contract with a tokenscript that has Russian/Chinese translation of every return messages or a contract author insist on returning a string because he doesn't want to depend on TokenScript.

Reply:
Note that the `as="bool"` boon proposed cannot be replaced by `syntax="[the oid of boolean'"` because it implies a hidden mapping table from a number to a string. If attestation is supplied, it will use the oid without mapping (i.e. use `TRUE` or `FALSE` coded in ASCII). In implementation, you can ignore this nuance until we have to actually implement attestations.

Reply:
Agreed. Input and output types should be explicitly. Booleans are just special here because we treat it as a supported type, just like when the syntax says “utf8”, the TokenScript client knows what to do. It doesn’t need a mapping here.

If we added a mapping tag in the TokenScript file when the syntax says Boolean (eg if it’s actually 0->true and non-zeros are error codes) then we use as=“unit” with a mapping tag.

Reply:
It's cleaner in the code to not use mapping for true/false, so this gets the approval from me.